From 227235ddce113b7d80aafbcfcd483c9200ffeab6 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Tue, 12 Jan 2016 17:34:57 +0800
Subject: [PATCH 0840/5242] MLK-12216 video: mipi_dsi: fix the potential
 uninitalized variable usage

commit  e8f0cb62945bcfedc06d2ef128e85dea05105362 from
https://source.codeaurora.org/external/imx/linux-imx.git

The 'val' variable may be used before uninitialized.
So initialize it at the begining.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/mipi_dsi.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/fbdev/mxc/mipi_dsi.c b/drivers/video/fbdev/mxc/mipi_dsi.c
index a62e6ba..3ebf25b 100644
--- a/drivers/video/fbdev/mxc/mipi_dsi.c
+++ b/drivers/video/fbdev/mxc/mipi_dsi.c
@@ -334,7 +334,7 @@ static void mipi_dsi_dphy_init(struct mipi_dsi_info *mipi_dsi,
 static void mipi_dsi_enable_controller(struct mipi_dsi_info *mipi_dsi,
 				bool init)
 {
-	u32		val;
+	u32		val = 0;
 	u32		lane_byte_clk_period;
 	struct  fb_videomode *mode = mipi_dsi->mode;
 	struct  mipi_lcd_config *lcd_config = mipi_dsi->lcd_config;
-- 
1.7.9.5

